app/models/draft_vaccination_record.rb

```diff
@@ -296,6 +293,31 @@ def reset_unused_fields
     end
   end
 
+  def earliest_possible_value
+    session.academic_year.to_academic_year_date_range.first.beginning_of_day
+  end
+
+  def latest_possible_value
+    [
+      session.academic_year.to_academic_year_date_range.last.end_of_day,
+      Time.current
+    ].min
+  end
+
+  def performed_at_within_range
+    if performed_at < earliest_possible_value
```

Comment:
I think it's possible for `performed_at` to be `nil` if the user has cleared the text boxes. They would still see a validation error from `validates :performed_at, presence: true` but I think this validation would fail due to the comparison of `nil`. Perhaps we can skip this check if it's `nil`?